builtin/credential/ldap/backend.go

```diff
@@ -106,7 +106,7 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 		sresult, err := c.Search(&ldap.SearchRequest{
 			BaseDN: cfg.UserDN,
 			Scope:  2, // subtree
-			Filter: fmt.Sprintf("(%s=%s)", cfg.UserAttr, EscapeLDAPValue(username)),
+			Filter: fmt.Sprintf("(%s=%s)", cfg.UserAttr, ldap.EscapeFilter(username)),
```

Comment:
I read RFC 4515 very carefully and I don't believe that's true. If you look at the RFC's BNF, you'll see that `assertionvalue` is the only value that is escaped. Looking at part of the BNF:
```
item           = simple / present / substring / extensible
simple         = attr filtertype assertionvalue
```
This defines a simple filter such as (cn=foobar). The item on the left is an `attr` which expands to `attributedescription` from RFC 4512, not an `assertionvalue`. I didn't find a reference in either RFC to a need to escape the `attr` value.

Feel free to convince me otherwise!